src/main/java/com/softserve/itacademy/controller/view/ViewController.java

```diff
@@ -1,34 +1,50 @@
 package com.softserve.itacademy.controller.view;
 
+import com.softserve.itacademy.security.perms.roles.AdminRolePermission;
+import com.softserve.itacademy.security.perms.roles.UserRolePermission;
 import org.springframework.http.MediaType;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
 
+import java.security.Principal;
+
 @Controller
 @RequestMapping
 public class ViewController {
-    
+
     @GetMapping(path = "/", produces = MediaType.TEXT_HTML_VALUE)
     public String homeView() {
         return "home.html";
     }
 
     @GetMapping(path = "/login", produces = MediaType.TEXT_HTML_VALUE)
-    public String loginView() {
-        return "login.html";
+    public String loginView(Principal principal) {
+        if (principal == null) {
+            return "login.html";
+        } else {
+            return "redirect:user";
+        }
     }
 
     @GetMapping(path = "/registration", produces = MediaType.TEXT_HTML_VALUE)
-    public String registrationView() {
-        return "registration.html";
+    public String registrationView(Principal principal) {
+        if (principal == null) {
+            return "registration.html";
+        } else {
+            return "redirect:user";
```

Comment:
Can we also check if it is admin registered or logged in, and redirect him to admin panel?

Reply:
We could, but I still think that we should redirect everybody to the last page user was at, or maybe to home page. Redirecting to panels sounds useless.